src/pages/index.js

```diff
@@ -20,58 +33,205 @@ const enableValidation = (settings) => {
 };
 
 //User Info
-const userinfo = new UserInfo(".profile__name", ".profile__subtitle");
+const userinfo = new UserInfo(".profile__name", ".profile__subtitle", "#profile-avatar");
 
 //Card
-function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleImageClick);
-  return cardElement.getCardElement();
+function createCard({name, link, isLiked, _id}) {
+  return new Card({name, link, isLiked, _id}, "#card-template", handleImageClick, 
+  handleTrashButtonClick, 
+  handleHeartButton,
+  ).getCardElement();
+ // const trashButton = cardElement.getTrashButton();
+ // trashButton.addEventListener("click", ()=> {
+   // popUpConfirm.open();
+ // });
+
+  //return cardElement.getCardElement();
 }
 
+
 //Section
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (item) => {
       const cardElement = createCard(item);
       section.addItem(cardElement);
     },
   },
-  ".cards__content"
+  ".cards__content",
 );
 section.renderItems();
 
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+//process the result
+.then(([cardData, formData]) => {
+ // console.log("Card Data:", cardData);
+ // console.log("Form Data:", formData)
+ userinfo.setUserInfo(formData);
+ userinfo.setAvatar(formData.avatar);
+ cardData.forEach((item) => {
+ 
+  //console.log("is like and unlike here at ForEach?",cardData)
+  //cardData.setLikeStatus(cardData.isLiked)
+  //const cardElement = createCard(item)
+   section.addItem(createCard(item));
+   //console.log("is like and unlike here at addItem???")
+ })
+})
+.catch((err) => {
+ console.error(err);// log the error to the console
+})
+.finally(() => {
+section.renderItems();
+//console.log("is like and unlike here at renderItems???", renderItems)
+}); 
+
 //Popups
+
+//PopUpWithImage
 const popUpWithImage = new PopUpWithImage("#modal-previewImage");
 popUpWithImage.setEventListeners();
 
+//PopUpWithForm
 const popUpEditProfile = new PopUpWithForm(
   "#modal-edit-profile",
-  (formData) => {
-    userinfo.setUserInfo(formData);
-    popUpEditProfile.close();
+  ({name, about}) => {
+    return new Promise((resolve, reject) => {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(closeAllPopups)
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```


You need to pass the **request** into the call of `handleSubmit` like that:

```js
 // here is an example
 function handleProfileFormSubmit(inputValues) {
  // here we create a function that returns a promise 
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return api.editProfile(inputValues).then(setCurrentUser);
  }
  // here we call handleSubmit passing the request
  handleSubmit(makeRequest);
}


```

So, this way you can remove a lot of code duplicating

Reply:
i'll definitely make a note of this and will try to incorporate it after solving the current issues with the code. as I don't want to complicate the refactoring. huge thank you though for this :)